integration/src/test/java/org/mule/test/integration/lifecycle/DependencyInjectionTestCase.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.integration.lifecycle;
+
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.mule.test.allure.AllureConstants.LifecycleAndDependencyInjectionFeature.ArtifactObjectsDependencyInjectionStory.ARTIFACT_OBJECTS_DEPENDENCY_INJECTION_STORY;
+import static org.mule.test.allure.AllureConstants.LifecycleAndDependencyInjectionFeature.LIFECYCLE_AND_DEPENDENCY_INJECTION;
+import org.mule.test.AbstractIntegrationTestCase;
+
+import org.junit.Test;
+import ru.yandex.qatools.allure.annotations.Features;
+import ru.yandex.qatools.allure.annotations.Stories;
+
+@Features(LIFECYCLE_AND_DEPENDENCY_INJECTION)
+@Stories(ARTIFACT_OBJECTS_DEPENDENCY_INJECTION_STORY)
+public class DependencyInjectionTestCase extends AbstractIntegrationTestCase {
+
+  @Override
+  protected String getConfigFile() {
+    return "org/mule/test/integration/lifecycle/artifact-object-dependency-injection-config.xml";
+  }
+
+  @Test
+  public void validateInjectedObjects() {
```

Comment:
you're free not to do this change if you want, but I think this would be a lot clearer if each of this assertions move to its own test method